addons/account/models/account_move.py

```diff
@@ -1203,11 +1203,13 @@ def date_key(move):
                 try:
                     if not move.posted_before:
                         move._constrains_date_sequence()
-                    # Has already a name or is not posted, we don't add to a batch
-                    continue
+                    # Has already a name or is not posted
                 except ValidationError:
-                    # Has never been posted and the name doesn't match the date: recompute it
-                    pass
+                    # Has never been posted and the name doesn't match the date: reset to draft
+                    move.name = '/'
```

Comment:
Shouldn't this only be the case if `move._get_last_sequence(lock=False)`?

Reply:
Hmm, yes, that makes sense. Otherwise it would also reset when switching to a period with no moves in it. I'll add the check.

Reply:
This would only go wrong when we are working on multiple records. Since higher up, we already check for that condition for the first record. The records after that will fail indeed.

```python
highest_name = self[0]._get_last_sequence(lock=False) if self else False

if not highest_name and move == self[0] and not move.posted_before and move.date:
    pass
```